src/mongocxx/client.hpp

```diff
@@ -183,8 +183,12 @@ class MONGOCXX_API client {
     cursor list_databases() const;
 
    private:
-    friend class database;
     friend class collection;
+    friend class database;
+    friend class pool;
+    friend class client_deleter;
```

Comment:
Do you think this is better than having the pool have a private non-exported "_release" method?